packages/analytics-js-plugins/src/xhrQueue/index.ts

```diff
@@ -1,14 +1,156 @@
 /* eslint-disable no-param-reassign */
-import { ExtensionPlugin, PluginName, ApplicationState } from '../types/common';
+import { QUEUE_NAME, REQUEST_TIMEOUT_MS } from './constants';
+import { XHRQueueItem } from './types';
+import {
+  getNormalizedQueueOptions,
+  getDeliveryPayload,
+  validatePayloadSize,
+  getDeliveryUrl,
+} from './utilities';
+import {
+  ExtensionPlugin,
+  PluginName,
+  ApplicationState,
+  DoneCallback,
+  HttpClient,
+  IErrorHandler,
+  ILogger,
+  QueueOpts,
+  RudderEvent,
+  IHttpClient,
+} from '../types/common';
+import { Queue, getCurrentTimeFormatted, toBase64 } from '../utilities/common';
 
 const pluginName = PluginName.XhrQueue;
 
+let eventsQueue: Queue;
+let writeKey: string;
+let dataplaneUrl: string;
+let httpClient: IHttpClient;
+let logger: ILogger | undefined;
+let errorHandler: IErrorHandler | undefined;
```

Comment:
I am a bit worried that this will cause issues in multible instances of the sdk exist and they have multible instances of the plugin

Reply:
I had in my mind that we would be passing ann required values as arguments to the extension point method call and these methods would be JS pure functions

Reply:
Addressed.